src/main/java/org/gridsuite/modification/server/dto/byfilter/equipmentfield/BatteryField.java

```diff
@@ -22,27 +22,27 @@ public enum BatteryField {
     REACTIVE_POWER_SET_POINT,
     DROOP;
 
-    public static Double getReferenceValue(Battery battery, String batteryField) {
+    public static String getReferenceValue(Battery battery, String batteryField) {
         ActivePowerControl<Battery> activePowerControl = battery.getExtension(ActivePowerControl.class);
         BatteryField field = BatteryField.valueOf(batteryField);
         return switch (field) {
-            case MINIMUM_ACTIVE_POWER -> battery.getMinP();
-            case MAXIMUM_ACTIVE_POWER -> battery.getMaxP();
-            case ACTIVE_POWER_SET_POINT -> battery.getTargetP();
-            case REACTIVE_POWER_SET_POINT -> battery.getTargetQ();
-            case DROOP -> activePowerControl != null ? activePowerControl.getDroop() : null;
+            case MINIMUM_ACTIVE_POWER -> String.valueOf(battery.getMinP());
+            case MAXIMUM_ACTIVE_POWER -> String.valueOf(battery.getMaxP());
+            case ACTIVE_POWER_SET_POINT -> String.valueOf(battery.getTargetP());
+            case REACTIVE_POWER_SET_POINT -> String.valueOf(battery.getTargetQ());
+            case DROOP -> activePowerControl != null ? String.valueOf(activePowerControl.getDroop()) : null;
         };
     }
 
-    public static void setNewValue(Battery battery, String batteryField, Double newValue) {
+    public static void setNewValue(Battery battery, String batteryField, String newValue) {
         BatteryField field = BatteryField.valueOf(batteryField);
         switch (field) {
-            case MINIMUM_ACTIVE_POWER -> battery.setMinP(newValue);
-            case MAXIMUM_ACTIVE_POWER -> battery.setMaxP(newValue);
-            case ACTIVE_POWER_SET_POINT -> battery.setTargetP(newValue);
-            case REACTIVE_POWER_SET_POINT -> battery.setTargetQ(newValue);
+            case MINIMUM_ACTIVE_POWER -> battery.setMinP(Double.parseDouble(newValue));
+            case MAXIMUM_ACTIVE_POWER -> battery.setMaxP(Double.parseDouble(newValue));
+            case ACTIVE_POWER_SET_POINT -> battery.setTargetP(Double.parseDouble(newValue));
+            case REACTIVE_POWER_SET_POINT -> battery.setTargetQ(Double.parseDouble(newValue));
             case DROOP -> battery.newExtension(ActivePowerControlAdder.class)
-                    .withDroop(newValue)
+                    .withDroop(Double.parseDouble(newValue))
                     .add();
         }
```

Comment:
YES! getReferenceValue and setNewValue in fact are equivalent to getter and setter for a field of an equipment. Because the data types may be different so it is better to untyping values then in function of data type known in the dto info, we can re-convert to the right type if needed. This is a similar solution as we did in the filter library.